gboard_autopsy.py

```diff
@@ -0,0 +1,123 @@
+import inspect
+import os
+
+from org.sleuthkit.autopsy.casemodule import Case
+from org.sleuthkit.autopsy.casemodule import NoCurrentCaseException
+from org.sleuthkit.autopsy.coreutils import Logger
+from org.sleuthkit.autopsy.ingest import IngestModuleFactoryAdapter
+from org.sleuthkit.autopsy.ingest import DataSourceIngestModule
+from org.sleuthkit.autopsy.ingest import IngestModule
+from org.sleuthkit.autopsy.ingest import IngestServices
+from org.sleuthkit.datamodel.blackboardutils import ArtifactsHelper
+from org.sleuthkit.datamodel import TskCoreException, TskDataException
+from org.sleuthkit.autopsy.casemodule.services import Blackboard
+from org.sleuthkit.autopsy.ingest.IngestModule import IngestModuleException
+
+from java.util.logging import Level
+
+class GboardDataSourceIngestModuleFactory(IngestModuleFactoryAdapter):
+
+    moduleName = "GBoard4A"
+
+    def getModuleDisplayName(self):
+        return self.moduleName
+
+    def getModuleDescription(self):
+        return "Extracts Android GBoard app data."
+
+    def getModuleVersionNumber(self):
+        return "1.0"
+
+    def isDataSourceIngestModuleFactory(self):
+        return True
+
+    def createDataSourceIngestModule(self, ingestOptions):
+        return GboardDataSourceIngestModule()
+
+class GboardDataSourceIngestModule(DataSourceIngestModule):
+
+    _logger = Logger.getLogger(GboardDataSourceIngestModuleFactory.moduleName)
+
+    GBOARD_CLIPBOARD_ARTIFACT = 'GBOARD_CLIPBOARD_OBJECT'
+    GBOARD_PACKAGE_NAME = 'com.google.android.inputmethod.latin'
```

Comment:
Should this be hardcoded?

Reply:
I don't see other options here

Reply:
How about when the input method isn't latin based?

Reply:
> How about when the input method isn't latin based?

GBoard is no longer `latin` based. This package name is basically for backward compatibility. There was keyboards for other input methods, but got deprecated. See: https://play.google.com/store/apps/details?id=com.google.android.inputmethod.japanese,  https://play.google.com/store/apps/details?id=com.google.android.inputmethod.pinyin and https://play.google.com/store/apps/details?id=com.google.android.apps.inputmethod.hindi .

Those keyboards are all deprecated and no longer updated and got merged into GBoard. If you check the description of the app here: https://play.google.com/store/apps/details?id=com.google.android.inputmethod.latin you'll find out that there are other input methods supported like the supported by the apps above described.

Even if we wanted to support that, if this was a thing, Google doesn't follow standards, for some reason: `com.google.android.apps.inputmethod.hindi` != `com.google.android.inputmethod.hindi`